src/Microsoft.DotNet.VersionTools/Dependencies/FileRegexUpdater.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.VersionTools.Dependencies
+{
+    public abstract class FileRegexUpdater : IDependencyUpdater
+    {
+        public string Path { get; set; }
+        public Regex Regex { get; set; }
```

Comment:
Just to make sure we're talking about the same regexes, the one here would be replacing the content of these elements: https://github.com/dotnet/corefx/blob/2574648/dir.props#L91-L93. So an example would be `<CoreFxExpectedPrerelease>(?<version>.*)<`. A strong model would be better, but at least this is pretty simple.

The package-matching regexes on the other hand are definitely bad, and https://github.com/dotnet/buildtools/issues/732 tracks changing to validate based on the versions repo package lists. (It sounds like you're talking about these with inclusions/exclusions, which I've had my fair share of problems with. :smile:)